### PR TITLE
fix(sql-rule): several sql interception bugs

### LINKED ISSUE #1159: [Bug]: SQL statements that violate SQL window guidelines should be intercepted.
### ODC version

ODC 4.2.3

### OB version

-

### What happened?

image
The SQL statement that violates the SQL window guidelines was not intercepted, and a database change task has been initiated.

### What did you expect to happen?

SQL statements that violate SQL window guidelines should be intercepted.

### How can we reproduce it (as minimally and precisely as possible)?

Executing SQL statements that violate SQL window guidelines.

### Anything else we need to know?

_No response_

### Cloud

_No response_

### PR DESCRIPTION
**Pull request by MarkPotato777**
#### What type of PR is this?

type-bug

#### What this PR does / why we need it:
This PR fixed several bugs:
- if the sql is intercepted by sql-console rules, then sql-check rules will not be checked;
- make the rule 'max-execute-sql-size' apply to the last sql;
- when the rule 'allow-execute-sql-type' value is empty, users could execute any type of sqls;
- when quering data, make sql-console rules not take effect;

#### Which issue(s) this PR fixes:

Fixes #1159 #1162 #1166

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by yhilmare** [comment]
we may merge this pr #1163 first, It looks like having some conflicts~

File: server/odc-service/src/main/java/com/oceanbase/odc/service/integration/ExternalSqlInterceptor.java [hunk 1]
```diff
@@ -109,7 +109,11 @@ public boolean doPreHandle(@NonNull SqlAsyncExecuteReq request, @NonNull SqlAsyn
                         .ifPresent(rule -> {
                             Rule violationRule = new Rule();
                             RuleViolation violation = new RuleViolation();
+                            violation.setText(request.getSql());
```
Comment (MarkPotato777) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/integration/ExternalSqlInterceptor.java:112]: if violates the external SQL interception rule, then put all SQLs into one line, and the highlight is all the SQLs.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/regulation/ruleset/SqlConsoleRuleService.java [hunk 2]
```diff
@@ -146,7 +146,7 @@ public <T> Optional<List<T>> getListProperties(@NonNull Long rulesetId, @NonNull
                 return result;
             }).collect(Collectors.toList());
             if (CollectionUtils.isEmpty(properties) || properties.get(0).isEmpty()) {
-                return Optional.empty();
+                return Optional.of(Collections.emptyList());
```
Comment (MarkPotato777) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/regulation/ruleset/SqlConsoleRuleService.java:149]: fix #1162

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**